ocs_ci/utility/powernodes.py

```diff
@@ -0,0 +1,132 @@
+
+    def iskvm(self):
+        """
+        Verify if this environment is libvirt or kvm.
+        Args: None
+        Returns:
+            bool: True if this is kvm environment, False otherwise
+        """
+        return self.isKVM
+
+    def verify_machine_is_down(self, node):
+        """
+        Verify if PowerNode is completely powered off
+
+        Args:
+            node (object): Node objects
+
+        Returns:
+            bool: True if machine is down, False otherwise
+
+        """
+        result = exec_cmd(
+            "sudo virsh domstate test-ocp" + get_ocp_version("-") + f"-{node.name}"
+        )
+        if result.stdout.lower().rstrip() == b"running":
+            return False
+        else:
+            return True
+
+    def stop_powernodes_machines(self, powernode_machines, force=True):
+        """
+        Stop PowerNode Machines
+
+        Args:
+            powernode_machines (list): PowerNode objects
+            force (bool): True for PowerNode ungraceful power off, False for
+                graceful PowerNode shutdown - for future use
+
+        Raises:
+            UnexpectedBehaviour: If PowerNode machine is still up
+
+        """
+        ocpversion = get_ocp_version("-")
+        for node in powernode_machines:
+            cmd = "sudo virsh shutdown test-ocp" + ocpversion + f"-{node.name}"
+            result = exec_cmd(cmd)
+            """
+            if result.returncode == 0:
+                return False
+            else
+                return True
+            """
+            logger.info(f"Result of shutdown {result}")
+            logger.info("Verifying node is down")
+            ret = TimeoutSampler(
+                timeout=900,
+                sleep=3,
+                func=self.verify_machine_is_down,
+                node=node,
+            )
+            logger.info(ret)
+            if not ret.wait_for_func_status(result=True):
+                raise UnexpectedBehaviour("Node {node.name} is still Running")
+
+    def start_powernodes_machines(self, powernode_machines, force=True):
+        """
+        Start PowerNode Machines
+
+        Args:
+            powernode_machines (list): List of PowerNode machines
+            wait (bool): Wait for PowerNodes to start - for future use
+
+        """
+        ocpversion = get_ocp_version("-")
+        for node in powernode_machines:
+            result = exec_cmd(
+                "sudo virsh start test-ocp" + ocpversion + f"-{node.name}"
+            )
+            logger.info(f"Result of shutdown {result}")
+
+        wait_for_cluster_connectivity(tries=900)
+        wait_for_nodes_status(
+            node_names=get_master_nodes(), status=constants.NODE_READY, timeout=900
+        )
+        wait_for_nodes_status(
+            node_names=get_worker_nodes(), status=constants.NODE_READY, timeout=900
+        )
+
+    def restart_powernodes_machines(self, powernode_machines, force=True):
+        """
+
```

Comment:
No need for the empty line at the start of the docstring

```suggestion
```

Reply:
Done.